src/zerocoin_v3.cpp

```diff
@@ -648,10 +642,15 @@ void CZerocoinStateV3::RemoveBlock(CBlockIndex *index) {
 			mintedPubCoins.erase(coinIt);
 		}
 	}
+
+	index->mintedPubCoinsV3.clear();
```

Comment:
Is variable index used anywhere else, after we remove the block?

Reply:
Yes it not destroy and it will be reuse later so that why it make state corrupted.

Reply:
Aha, so it's not cleared, and it can be used only in the case where we RETURN to this branch of the blockchain right? In that case this index is re-used, and we need to clean it up. 

Can you add a long comment to the code describing it? Because if someone want to read this code in the future, that would help a lot. It was not obvious to me at all.

Reply:
Okay good idea. I will add comment here.